examples/server/server.cpp

```diff
@@ -571,17 +547,22 @@ struct server_response {
         waiting_task_ids.erase(id_task);
     }
 
-    // This function blocks the thread until there is a response for this id_task
-    server_task_result recv(int id_task) {
+    void remove_waiting_tasks(std::vector<server_task> & tasks) {
+        for (const auto & t : tasks) {
+            remove_waiting_task_id(t.id);
+        }
+    }
+
+    // This function blocks the thread until there is a response for one of the id_tasks
+    server_task_result recv(std::unordered_set<int> & id_tasks) {
```

Comment:
```suggestion
    server_task_result recv(const std::unordered_set<int> & id_tasks) {
```

Reply:
This change was added in https://github.com/ggerganov/llama.cpp/pull/9274/commits/fa2210652af7dea17e404f038c38efd745f5186a . I discovered that not having `const` will requires lambda function to be `mutable`, which cause double-free in [this CI run](https://github.com/ggerganov/llama.cpp/actions/runs/10666482802/job/29562005029) (I'm not sure why, but good to know!)